### PR TITLE
BUG: dask.array: asarray(..., copy=None) copies in dask==2024.12

### PR DESCRIPTION
**Pull request by ev-br**
closes gh-209

### CODE REVIEW

**Comment by ev-br** [comment]
Two issues:
- on python 3.9 we install an earlier dask version, which does not have a breaking change
- array-api-tests hang, https://github.com/data-apis/array-api-compat/issues/210

**Comment by betatim** [comment]
In [the standard for `asarray`](https://data-apis.org/array-api/draft/API_specification/generated/array_api.asarray.html#array_api.asarray) it says that for `copy=None` you should not copy if possible. Do you think "in dask we always make a copy" is compatible with that? And if no, is there anything we can do in `array-api-compat` to fix this? It would be nice if we could organise (through the compat layer) for users to get the same behaviour no matter what version of dask they are using.

If there is nothing we can do to fix this then I think updating the test is fine.

**Comment by lucascolley** [comment]
Yes, I think it is compatible for Dask to say that they will always copy, and just raise an error for `copy=False`.

**Comment by ev-br** [comment]
The issue with 2024.12 is `copy=None`, actually.

**Comment by lucascolley** [comment]
> The issue with 2024.12 is copy=None, actually.

It is compatible for them to always copy with `copy=None`.

**Comment by ev-br** [comment]
CI is green with three tweaks
- asarray(..., copy=None) for array-likes always copies
- the test for copy=None is updated to match
- a unrelated test needs to be skipped on CI: was xfailed and was hanging with dask 2024.12

File: tests/test_common.py [hunk 1]
```diff
@@ -204,8 +204,10 @@ def test_asarray_copy(library):
     b = asarray(a, copy=None)
     assert is_lib_func(b)
     a[0] = 0.0
-    if library == 'cupy':
+    if library in ('cupy', 'dask.array'):
         # A copy is required for libraries where the default device is not CPU
+        # dask made a breaking change in 2024.12: copy=None copies
```
Comment (crusaderky) [inline tests/test_common.py:209]:
```suggestion
        # dask changed behaviour of copy=None in 2024.12 to copy
```
IMHO not a breaking change

**Review by crusaderky — APPROVED** [review]

**Review by lucascolley — APPROVED** [review]

**Comment by ev-br** [comment]
Let's unblock everything then :-)

**Review by crusaderky — COMMENTED** [review]
See also: https://github.com/data-apis/array-api-compat/pull/235

**Pull request merged**